operator/v1/types_ingress.go

```diff
@@ -1346,6 +1346,78 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// For Setting Headers you can provide a value for the header whereas for deleting one needs to provide header name.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActions defines whether we want to set/delete Response or Request Headers.
+type IngressControllerHTTPHeaderActions struct {
+	// Response is a list of http response headers which either need to set or deleted.
+	Response []IngressControllerHTTPHeader `json:"response,omitempty"`
+	// Request is a list of http request headers which either need to set or deleted.
+	Request []IngressControllerHTTPHeader `json:"request,omitempty"`
```

Comment:
but we have mentioned this in the enhancement right ? it will become a big document. I am not sure why we need to provide so many details here. May be for user to read via oc explain ?

Reply:
Right, this is end-user documentation.

Reply:
Miheer and I discussed this, and agreed that the API godoc should document how the various API fields (`spec.httpHeaders.actions`, `spec.clientTLS`, `spec.httpHeaders.forwardedHeaderPolicy`, `spec.httpHeaders.uniqueId`, and `spec.httpHeaders.headerNameCaseAdjustments`) interact with each other, and product documentation can cover how `spec.httpHeaders.actions` interacts with session cookies and `cache-control`.